### PR TITLE
Feature mes 2368 unauthenticated logging

### PR DESCRIPTION
**Pull request by rossfauldsbjss**
MES - 2368 log unauthenticated requests - e.g user cancelled sign in, or failed authorisation.
Logging service was changed by another request to use API key authentication instead of the custom authoriser using JWT.

logging API key, URL and interval are now going to be provided by MDM/airwatch. 
As the app cant access remote config if the user is unauthenticated.
Separate piece of work will remove logging parameters from remote config.

## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [ ] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [x] PR link added to JIRA
- [x] Reviewers selected in Github
- [ ] Any new reusable component/widget added to component library on Confluence

### CODE REVIEW

File: src/environment/environment.dev.ts [hunk 1]
```diff
@@ -7,6 +7,8 @@ export const environment: EnvironmentFile = {
   daysToCacheLogs: 7,
   logoutClearsTestPersistence: true,
   logsPostApiKey: '',
+  logsApiUrl: 'https://john.mes.dev-dvsacloud.uk/v1/logs',
```
Comment (johnmorleybjss) [inline src/environment/environment.dev.ts]: Using my API subdomain for quite a few envs. I won't block as I'm out this afternoon, but please fix!

Comment (rossfauldsbjss) [reply src/environment/environment.dev.ts]: replaced john with dev in two envs.

File: src/environment/environment.production.ts [hunk 2]
```diff
@@ -6,6 +6,8 @@ export const environment: EnvironmentFile = {
   daysToCacheJournalData: 7,
   daysToCacheLogs: 7,
   logsPostApiKey: '',
+  logsApiUrl: 'https://prod.mes.dev-dvsacloud.uk/v1/logs',
```
Comment (rstepnowski) [inline src/environment/environment.production.ts]: The URL is incorrect for PROD but the bigger issue here is that we probably don't want to put URLs to prod env in github at all.

Given the fact that this info comes form MDM anyways - I'd prefer to leave this empty - at least for prod/uat/perf/staging(?) - is there any value of hardcoding those URLs on higher ENVs than local/dev?
If we really need those URLs documented somewhere confluence would be a better place for it imho.

Comment (rossfauldsbjss) [reply src/environment/environment.production.ts]: I can remove this (and will)   - there is a url in there for the config service already though.

File: src/pages/login/login.ts [hunk 3]
```diff
@@ -195,4 +217,13 @@ export class LoginPage extends BasePageComponent {
       this.loadingSpinner = null;
     }
   }
+
+  private createLog(logType: LogType, actionType: string, message: string): Log {
```
Comment (jameshighfield) [inline src/pages/login/login.ts]: Unused param `actionType`. Did you intend on adding this to the log?

Comment (rossfauldsbjss) [reply src/pages/login/login.ts]: Great spot. Unintentional. Removing superfluous parameter now.

File: src/pages/login/login.ts [hunk 4]
```diff
@@ -106,8 +109,27 @@ export class LoginPage extends BasePageComponent {
...
         this.analytics.logException(error, true);
+        const log: Log = this.createLog(LogType.INFO,
+          `user cancelled login`);
+        this.store$.dispatch(new SaveLog(log));
+        this.store$.dispatch(new SendLogs());
       }
       if (error === AuthenticationError.USER_NOT_AUTHORISED) {
+
+        const token = await this.authenticationProvider.getAuthenticationToken();
+        const examiner = await this.authenticationProvider.getEmployeeId() || '';
```
Comment (ammarhaiderbjss) [inline src/pages/login/login.ts]: Perhaps change this to say 'unavailable' so the log is more intelligible?

File: src/pages/login/login.ts [hunk 5]
```diff
@@ -106,8 +109,27 @@ export class LoginPage extends BasePageComponent {
 
       if (error === AuthenticationError.USER_CANCELLED) {
         this.analytics.logException(error, true);
+        const log: Log = this.createLog(LogType.INFO,
```
Comment (ammarhaiderbjss) [inline src/pages/login/login.ts]: Looks like we're repeating this a few times. Perhaps a function to handle the creation of log and dispatching called `saveAndDispatchLog` that accepts a message as a param?

File: src/pages/login/login.ts (hunk 4 above)
Comment (rossfauldsbjss) [reply src/pages/login/login.ts]: changed.

Comment (rossfauldsbjss) [reply src/pages/login/login.ts]: done

File: src/pages/login/login.ts (hunk 5 above)
Comment (rossfauldsbjss) [reply src/pages/login/login.ts]: done

**Pull request merged**